airflow/models/dagrun.py

```diff
@@ -931,3 +945,13 @@ def schedule_tis(self, schedulable_tis: Iterable[TI], session: Session = NEW_SES
             )
 
         return count
+
+    @provide_session
+    def get_log_filename_template(self, *, session: Session = NEW_SESSION) -> Optional[str]:
+        if self.log_filename_id is None:  # DagRun created before LogFilename introduction.
+            template = session.query(LogFilename.template).order_by(LogFilename.id).limit(1).scalar()
+        else:
+            template = session.query(LogFilename.template).filter_by(id=self.log_filename_id).one_or_none()
```

Comment:
```suggestion
            template = session.query(LogFilename.template).get(self.log_filename_id)
```

This shouldn't ever return None because of the RESTRICT FK.

Reply:
Neither of the branches should ever return None, in fact, because we always ensure there’s a row on startup. I’ll change this to raise an error if None is returned, that makes things more explicit.

Reply:
Turns out `get()` cannot be used here: `get() can only be used against a single mapped class`.

Reply:
Oh get only works on objects, but we are asking for a single column. TIL.